app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
Here's what happened when I replaced `underscore.png` with another PNG file:

![image](https://cloud.githubusercontent.com/assets/963826/8708657/3061c23e-2af2-11e5-9620-3c500d15c40c.png)

The file sizes also do not match. Correct: 275207 bytes. New: 275205. You can see the diff of the hexdump is also different:

![image](https://cloud.githubusercontent.com/assets/963826/8708689/6981a746-2af2-11e5-9cd8-dd663c850db6.png)